jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -372,6 +376,84 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         managerState.flags[StateFlags.IS_ONGOING] = false
     }
 
+    /**
+    helper function to generate 1p dependency metrics
+    makes GET requests to maven central to check for 1p dependencies for a small subset of the project's
+    dependencies
+     */
+    private fun makeGETRequests(groupId: String, artifactId: String, version: String, jobId: JobId): Boolean {
+        try{
+            val response = getTextFromUrl("https://search.maven.org/solrsearch/select?q=g:$groupId%20AND%20a:$artifactId%20AND%20v:$version%20AND%20p:jar&rows=20&wt=json")
```

Comment:
i'd also like to reiterate the same comment i (and our appsec engineer) had:

why can't we figure this out on the service side?
it seems like we're trying to push a lot of business logic to the client side which in turn needs to be duplicated across all the client implementations

Reply:
There are future enhancements based on the data from this metric that will be done on the IDE side, which is why this logic is being done on the client side. If you'd like to see specifics, I can share our roadmap with you internally.